src/server/main.py

```diff
@@ -1,30 +1,259 @@
+            return jsonify({"message": "Token is revoked!"}), 401
+
+        # Decode the token and retrieve the information contained in it
+        try:
+            data = jwt.decode(token, app.config["SECRET_KEY"], algorithms=["HS256"])
+
+            try:
+                connection: Connection = getDatabaseConnection()
+                sql: str = "SELECT uuid, email FROM users WHERE uuid=?"
+                cursor: Cursor = connection.cursor()
+                cursor.execute(sql, (data["uuid"],))
+                user_information: list = cursor.fetchall()
+                connection.close()
+            except:
+                return jsonify({"message": "Database connection error!"}), 500
+
+            unique_id: str = ""
+            email: str = ""
+
+            if user_information:
+                # Retrieve user uuid
+                unique_id = user_information[0][0]  # uuid
+                # Retrieve user email
+                email = user_information[0][1]  # email
+        except:
+            return jsonify({"message": "Token is invalid !"}), 401
+
+        # Returns the current logged-in users context to the routes
+        return fun(unique_id, email, *args, **kwargs)
+
+    return decorated
 
-@app.route("/logout")
-def logout() -> str:
-    """Logout endpoint."""
-    return "logout"
 
+def revoke_token(token: str) -> Any:
+    """Revoking activated token."""
+    decoded_token: str = jwt.decode(token, app.config["SECRET_KEY"], algorithms=["HS256"])
+    expiry: datetime = datetime.fromtimestamp(float(decoded_token["exp"]))
 
-@app.route("/register")
-def register() -> str:
+    # Saving revoked token into the database
+    try:
+        connection: Connection = getDatabaseConnection()
+        connection.execute(
+            "INSERT INTO revoked_tokens (token, expiry) VALUES (?, ?)",
+            (
+                token,
+                expiry,
+            ),
+        )
+        connection.commit()
+        connection.close()
+    except:
+        return jsonify({"message": "Database connection error!"}), 500
+
+
+def is_token_revoked(token: str) -> Any:
+    """Check if token is revoked."""
+    try:
+        connection: Connection = getDatabaseConnection()
+        result = connection.execute(
+            "SELECT token FROM revoked_tokens WHERE token=? AND expiry > ?",
+            (
+                token,
+                datetime.utcnow(),
+            ),
+        )
+        is_revoked: bool = result.fetchone() is not None
+        connection.close()
+    except:
+        return jsonify({"message": "Token is invalid !"}), 401
+
+    return is_revoked
+
+
+@app.route("/register", methods=["POST"])
+def register() -> Response:
     """Registration endpoint."""
-    return "register"
+    new_user: Any = request.json
+
+    unique_id: UUID = uuid.uuid4()
+    email: str = new_user["email"]
+    password: str = new_user["password"]
+    repeated_password: str = new_user["repeatedPassword"]
```

Comment:
To zabezpieczenie (repeated_password) raczej powinno być wyłącznie po stronie frotnendu - jest po to, aby upewnić się, że użytkownik poprawnie wpisał hasło.